### ISSUE TITLE
Implement interior face culling for mesh generation

### ISSUE THREAD

**Issue by quincr** [DuckyShine004/opengl-voxel-engine#56]
This might reduce the initial load-in time.

**Comment by DuckyShine004** [comment]
Good idea, I was thinking the same thing, need to take out the geometry shader as well.

**Comment by qcr86** [comment]
This might be able to be done using a VBO per chunk, though that may require a texture atlas instead of array. I'm not exactly sure if that is faster, as in a personal project of mine, I am struggling to get good performance using 32x32x32 chunks, due to me not seeming to find a good way to extend vertices to a list efficiently ( for interior face culling. I don't use shaders in the project, so i'm not sure if it is faster to do so ).
